### DISCUSSION TITLE
Home Assistant ZHA integration (and zigpy) device support without custom ZHA-quirks (ZHA Device Handlers)?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#3] [General]
@ffenix113 FYI, I think you should be interested in joining and contributing to this zigpy discussion about "Quirks V2" as will make it easier for users of your project:

https://github.com/zigpy/zigpy/discussions/1312 ("**_[RFC] Quirks v2_**").

Above is a request for change to enable zigpy library to allow for "generic" device handlers / quirks which should make it easier to add support for non-standard Zigbee devices to Home Assistant's ZHA integration, which is something that can be a pain point for users of DIY and devices with odd features and functions:

* https://www.home-assistant.io/integrations/zha
  * https://www.home-assistant.io/integrations/zha#knowing-which-devices-are-supported
    * https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices

As I understand the idea/concept is very similar to https://github.com/Koenkk/zigbee2mqtt/pull/20358 (continuation of https://github.com/Koenkk/zigbee2mqtt/pull/20173) for Z2M.

I believe the main device is that custom quirks (device handlers/converters/parsers/translators) will still be needed for ZHA inside the [ZHA Device Handlers](https://github.com/zigpy/zha-device-handlers) repository, (which is updated very often in Home Assistant via "zha-quirks" on PyPi), but at least users do not need to update the zha component code inside Home 
Assistant code.

PS: Even if that zigpy RFC with "Quirks V2" for ZHA is not implemented you might still be interested in improving this quirk-generator: 

https://github.com/zigpy/quirk-generator

https://github.com/zigpy/quirk-generator/discussions

**Comment by ffenix113** [comment]
Hello Hedda,
Thank you for the information, but I don’t think that at this point in time I might have anything useful to add there.
But I will keep it in mind if situation changes.

Thank you!
